packages/core/src/geometry/GeometrySystem.js

```diff
@@ -586,22 +586,48 @@ export default class GeometrySystem extends System
      */
     draw(type, size, start, instanceCount)
     {
-        const { gl } = this;
+        const { gl, renderer: { context } } = this;
         const geometry = this._activeGeometry;
 
         // TODO.. this should not change so maybe cache the function?
 
         if (geometry.indexBuffer)
         {
-            if (geometry.instanced)
+            const byteSize = geometry.indexBuffer.data.BYTES_PER_ELEMENT;
+
+            if (byteSize === 2)
+            {
+                if (geometry.instanced)
+                {
+                    /* eslint-disable max-len */
+                    gl.drawElementsInstanced(type, size || geometry.indexBuffer.data.length, gl.UNSIGNED_SHORT, (start || 0) * byteSize, instanceCount || 1);
```

Comment:
And this could become:

```gl.drawElementsInstanced(type, size || geometry.indexBuffer.data.length, glType, (start || 0) * byteSize, instanceCount || 1);```

Reply:
But also I need some where to check whether I can use uint32.
https://github.com/pixijs/pixi.js/pull/5738/files#diff-16f6fe0290c560188da725c2135c1eaaR613